project/Build.scala

```diff
@@ -223,36 +228,76 @@ object Build extends sbt.Build {
       )
   ) dependsOn (streaming % "test->test; provided", dsl % "provided")
 
-  lazy val services = Project(
-    id = "gearpump-services",
-    base = file("services"),
-    settings = commonSettings  ++ 
-      Seq(
-        libraryDependencies ++= Seq(
-          "io.spray" %%  "spray-testkit"   % sprayVersion % "test",
-          "io.spray" %%  "spray-httpx"     % sprayVersion,
-          "io.spray" %%  "spray-client"    % sprayVersion,
-          "io.spray" %%  "spray-json"    % sprayJsonVersion,
-          "com.wandoulabs.akka" %% "spray-websocket" % sprayWebSocketsVersion
-            exclude("com.typesafe.akka", "akka-actor_2.11"),
-          "org.webjars" % "angularjs" % "1.4.3", // We stick on 1.3.x until angular-ui-select works
-          "org.webjars" % "bootstrap" % "3.3.5",
-          "com.lihaoyi" %% "upickle" % upickleVersion,
-          "org.webjars" % "d3js" % "3.5.5",
-          "org.webjars" % "momentjs" % "2.10.3",
-          "org.webjars.bower" % "angular-loading-bar" % "0.8.0",
-          "org.webjars.bower" % "angular-smart-table" % "2.1.1",
-          "org.webjars.bower" % "angular-motion" % "0.4.2",
-          "org.webjars.bower" % "bootstrap-additions" % "0.3.1",
-          "org.webjars.bower" % "angular-strap" % "2.3.1",
-          "org.webjars.bower" % "angular-ui-select" % "0.12.0",
-          "org.webjars" % "select2" % "3.5.2",
-          "org.webjars" % "select2-bootstrap-css" % "1.4.6",
-          "org.webjars.bower" % "ng-file-upload" % "5.0.9",
-          "org.webjars.bower" % "vis" % "4.5.1"
-        ).map(_.exclude("org.scalamacros", "quasiquotes_2.10")).map(_.exclude("org.scalamacros", "quasiquotes_2.10.3"))
-      )
-  ) dependsOn(streaming % "test->test; compile->compile", daemon % "test->test; compile->compile")
+  lazy val services = Project(id = "gearpump-services", base = file("services")).
+    settings(commonSettings: _*).
+    aggregate(servicesjs, servicesjvm).
```

Comment:
Correct. But I did not get fastOptJS to work. Having serviceJVM depend on fastOptJS in serviceJS results in a recursion error. The result is that fastOptJS is never invoked.



> On Aug 3, 2015, at 8:28 PM, Sean Zhong <notifications@github.com> wrote:
> 
> In project/Build.scala:
> 
> > -          "org.webjars.bower" % "angular-loading-bar" % "0.8.0",
> > -          "org.webjars.bower" % "angular-smart-table" % "2.1.1",
> > -          "org.webjars.bower" % "angular-motion" % "0.4.2",
> > -          "org.webjars.bower" % "bootstrap-additions" % "0.3.1",
> > -          "org.webjars.bower" % "angular-strap" % "2.3.1",
> > -          "org.webjars.bower" % "angular-ui-select" % "0.12.0",
> > -          "org.webjars" % "select2" % "3.5.2",
> > -          "org.webjars" % "select2-bootstrap-css" % "1.4.6",
> > -          "org.webjars.bower" % "ng-file-upload" % "5.0.9",
> > -          "org.webjars.bower" % "vis" % "4.5.1"
> > -        ).map(_.exclude("org.scalamacros", "quasiquotes_2.10")).map(_.exclude("org.scalamacros", "quasiquotes_2.10.3"))
> > -      )
> > -  ) dependsOn(streaming % "test->test; compile->compile", daemon % "test->test; compile->compile")
> > +  lazy val services = Project(id = "gearpump-services", base = file("services")).
> > +    settings(commonSettings: _*).
> > +    aggregate(servicesjs, servicesjvm).
> I think cross-project shall work. For cross-project, there are still project created.
> One is services.js, the other is services.jvm.
> 
> In your current code, you are operating on project servicesjs. After enabling scalajs, the only difference is that you are operating on project "service.js" (note the additonal dot ".")
> 
> —
> Reply to this email directly or view it on GitHub.
>